devito/ops/compiler.py

```diff
@@ -63,33 +58,28 @@ def _translate_ops(self, soname, ccode, hcode):
         c_file.close()
         h_file.close()
 
-        if self._ops_install_path:
-            # Calling OPS Translator
-            translator = '%s/../ops_translator/c/ops.py' % (self._ops_install_path)
-            subprocess.run([translator, c_file.name], cwd=self.get_jit_dir())
-        else:
-            warning("Couldn't find OPS_INSTALL_PATH \
-                environment variable, please check your OPS installation")
+        # Calling OPS Translator
+        translator = '%s/../ops_translator/c/ops.py' % (self._ops_install_path)
+        translation = subprocess.run([translator, c_file.name], cwd=self.get_jit_dir())
+        if translation.returncode == 1:
+            raise ValueError("OPS Translation Error")
 
-    def _compile_cuda(self, soname):
+    def _compile(self, soname):
         # CUDA kernel compilation
         cuda_src = '%s/CUDA/%s_kernels.cu' % (self.get_jit_dir(), soname)
         cuda_target = '%s/%s_kernels_cu' % (self.get_jit_dir(), soname)
 
-        cuda_code = ""
         try:
             with open(cuda_src, 'r') as f:
                 cuda_code = f.read()
         except FileNotFoundError:
             raise ValueError("Couldn't find file: %s" % cuda_src)
```

Comment:
It can still get better (for currently or future implementations).
cuda_src, cuda_target, etc. should stay in OPSCompilerCUDA. The same goes for the others (seq_src, openmp_src, etc).

Reply:
yes, I agree. It's indeed a violation that concepts of the subclass appear in the superclass